### ISSUE TITLE
Adversarial learning with keras

### ISSUE THREAD

**Issue by SqrtPapere** [keras-team/keras#9102]
Hello, I am trying to implement adversarial learning for sentiment analysis using a CNN. 

I already trained a model (without adversarial), what should I do now? I know I have to re-use the prediction of this model for training another model using modified samples.

Can you suggest some repositories or anything at all that can help me?

Any hint is very appreciated. Thank you!

**Comment by gabrieldemarmiesse** [comment]
https://github.com/keras-team/keras/blob/master/examples/mnist_acgan.py

This should help you. Also be aware of this common trap: using `layer.trainable = False` does nothing unless the model is compiled after changing this trainable attribute.

**Comment by SqrtPapere** [comment]
Thank you very much but I don't think this is what I need. I am starting to think GAN and adversarial learning are slightly different. 

I am trying to do what they do in this paper: [link](https://nlp.stanford.edu/courses/cs224n/2015/reports/20.pdf)

I need to calculate the loss on a train batch where the samples in the batch are summed by the gradient of the loss on the same batch not modified. In formula:

![screen shot 2018-01-22 at 11 18 48](https://user-images.githubusercontent.com/17984266/35216028-7bc6c5ee-ff66-11e7-8a39-dafb540f1b7b.png)


In order to do this I need to know the loss on the original example (or batch) and then add the gradient to the same batch. 

Maybe GAN are the same and I am losing something crucial. Thank you
